kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerManagementIntegrationTest.java

```diff
@@ -558,12 +559,71 @@ public void testUpdateContainerConfigOnNotExistingContainer() {
         }
     }
 
+    @Test
+    public void testTemplateKeyChangeDuringUpdate() {
+        ServerTemplate serverTemplate = createServerTemplate();
+
+        ContainerSpec containerToDeploy = new ContainerSpec(CONTAINER_ID, CONTAINER_NAME, serverTemplate, releaseId, KieContainerStatus.STOPPED, Collections.EMPTY_MAP);
+        mgmtControllerClient.saveContainerSpec(serverTemplate.getId(), containerToDeploy);
+
+        // Get container using kie controller.
+        ContainerSpec containerResponseEntity = mgmtControllerClient.getContainerInfo(kieServerInfo.getServerId(), CONTAINER_ID);
+        checkContainer(containerResponseEntity, KieContainerStatus.STOPPED);
+        assertEquals(CONTAINER_NAME, containerResponseEntity.getContainerName());
+        assertEquals(releaseId, containerResponseEntity.getReleasedId());
+
+        // setting new template to container
+        ServerTemplate secondTemplate = createServerTemplate("st-id", "st-id", kieServerInfo.getLocation());
+        containerToDeploy.setServerTemplateKey(secondTemplate);
+
+        try {
+            mgmtControllerClient.updateContainerSpec(serverTemplate.getId(), containerToDeploy);
+            fail("Template key should not be allowed to be changed during update.");
+        } catch (UnexpectedResponseCodeException e) {
+            assertEquals(400, e.getResponseCode());
+            KieServerAssert.assertResultContainsString(e.getMessage(), "Cannot change container template key during update.");
+        }
+
+        assertEquals(2, mgmtControllerClient.listServerTemplates().size());
+        // Check that on other server template is not any container.
+        KieServerAssert.assertNullOrEmpty("Found container in second server template.", mgmtControllerClient.listContainerSpec(secondTemplate.getId()));
+    }
+
+    @Test
+    public void testUpdateContainerServerTemplateId() {
```

Comment:
Please delete this test, there is nothing tested. Probably it is duplication of testTemplateKeyChangeDuringUpdate.